### PR TITLE
[Oracle Module] Refactor module to properly use host parsers instead of doing its own parsing of hosts

### LINKED ISSUE #31611: [Oracle Module] Refactor module to properly use host parsers instead of doing its own parsing of hosts
### Problem

Oracle Metricbeat Module currently does its [own parsing of hosts](https://github.com/elastic/beats/blob/main/x-pack/metricbeat/module/oracle/connection.go#L40) instead of using host parsers. Metricsets should rely on the host available in HostData.
For example,
If we look to the postgresql module, the url parser receives a [single url](https://github.com/elastic/beats/blob/ac7071bcf36233f216d12dd7473a20aacb6153ea/metricbeat/module/postgresql/postgresql.go#L56) to parse, and returns a single HostData, that contains a single host and is used like this by the metricset ([here](https://github.com/elastic/beats/blob/ac7071bcf36233f216d12dd7473a20aacb6153ea/metricbeat/module/postgresql/metricset.go#L47)), without having to check the raw host configuration.

### Solution

Metricbeat should instantiate one metricset per host, and the metricset shouldn't need to do any special handling to support multiple hosts. This module needs to be refactored to properly use host parsers instead of doing its [own parsing of hosts](https://github.com/elastic/beats/blob/main/x-pack/metricbeat/module/oracle/connection.go#L40). Using HostData is the only way a metricset instance has to know what host to use when multiple hosts are configured.

### Use cases

- When parsing the configured host, it allows to set in the HostData the Host, that is later used for service.address.
- It abstracts the handling of multiple hosts, each metricset instance receives a single HostData with the host they have to use.

### Related Converations

- https://github.com/elastic/beats/pull/31368#discussion_r867761693
- https://github.com/elastic/beats/pull/31368#discussion_r867762959

### PR DESCRIPTION
**Pull request by yug-elastic**
- Enhancement

## What does this PR do?

Refactors the Oracle module to properly use host parsers instead of doing its own parsing of hosts.

## Why is it important?

Metricbeat should instantiate one metricset per host, and the metricset shouldn't need to do any special handling to support multiple hosts. Using `HostData` is the only way a metricset instance has to know what host to use when multiple hosts are configured. When parsing the configured host, it allows to set in the `HostData` the Host, that is later used for `service.address`. It abstracts the handling of multiple hosts, each metricset instance receives a single `HostData` with the host they have to use.

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
~~- [ ] I have made corresponding changes to the documentation~~
~~- [ ] I have made corresponding change to the default configuration files~~
~~- [ ] I have added tests that prove my fix is effective or that my feature works~~
- [x] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

## Related issues

- Closes #31611

### CODE REVIEW

File: x-pack/metricbeat/module/oracle/connection.go [hunk 1]
```diff
@@ -6,62 +6,54 @@ package oracle
...
-// HostParser parsers the host value as a URL
-var HostParser = parse.URLHostParserBuilder{
-	DefaultScheme: "oracle",
-}.Build()
-
-func init() {
-	// Register the ModuleFactory function for the "oracle" module.
-	if err := mb.Registry.AddModule("oracle", newModule); err != nil {
-		panic(err)
+// HostParser parses host value
```
Comment (kush-elastic) [inline x-pack/metricbeat/module/oracle/connection.go:23]: Could you please add a more detailed comment here?
like `HostParser` parses host and other information from `connectionString` and giving example of `connectionString` and exact output.

**Review by kush-elastic — APPROVED** [review]
Just a small comment.
Rest LGTM!
Please wait for the others approvals.

**Comment by agithomas** [comment]
Please revisit the testing.go.
 Please refer other modules such as mysql / cockroachdb.

**Comment by yug-elastic** [comment]
> Please revisit the testing.go. Please refer other modules such as mysql / cockroachdb.

As a part of this PR, we are refactoring the host parsing mechanism of the Oracle Module, and based on our analysis, there does not seem to be a change required in the tests with respect to the changes we have made in the module. The `testing.go` ([link](https://github.com/elastic/beats/blob/main/x-pack/metricbeat/module/oracle/testing.go)) file of the module was developed along with the previous implementation (in which a single host is being parsed as expected) and the same tests are passing with respect to the changes in this PR.

**Review by agithomas — APPROVED** [review]
Looks good to me.

**Pull request merged**